plugins/BEdita/API/src/Controller/ObjectsController.php

```diff
@@ -502,6 +507,53 @@ public function relationships()
         return null;
     }
 
+    /**
+     * Sort relationships data for an object.
+     * This action replaces the related data with a sorted set of data.
+     * Data is sort by a field in ascending or descending order.
+     * The field and the direction are passed in the request meta data.
+     *
+     * Example:
+     * ```
+     * PATCH /{object_type}/{id}/relationships/{relationship}/sort
+     * {
+     *     "meta": {
+     *         "field": "{{ field }}",
+     *         "direction": "{{ direction }}"
+     *     }
+     * }
+     * ```
+     *
+     * @return \Cake\Http\Response|null
+     */
+    public function relationshipsSort(): ?Response
+    {
+        $this->request->allowMethod(['patch']);
+        try {
+            $id = $this->request->getParam('id');
+            $relationship = $this->request->getParam('relationship');
+            $payload = json_decode((string)$this->request->getBody(), true);
+            $field = (string)Hash::get($payload, 'meta.field');
+            $direction = (string)Hash::get($payload, 'meta.direction');
+            $association = $this->findAssociation($relationship);
+            $this->setRelationshipsAllowedMethods($association);
+            $action = new GetObjectAction(['table' => $this->Table, 'objectType' => $this->objectType]);
+            $entity = $action(['primaryKey' => $id]);
+            $action = new SortRelatedObjectsAction(compact('association'));
+            $count = $action(['entity' => $entity, 'field' => $field, 'direction' => $direction]);
+        } catch (\Exception $e) {
+            throw new InternalErrorException(
+                __d('bedita', 'Could not sort and update relationship "{0}": {1}', $relationship, $e->getMessage())
+            );
+        }
```

Comment:
I would avoid "try catch" here and I would leave that the actions decide which error thrown and therefore which status code for the response.

For example if `GetObjectAction` fail to find the object the action will throw a `RecordNotFoundException` and the response will be a `404 Not Found`.

The same with `SortRelatedObjectsAction`. If missing some required data it throws an `InvalidDataException` that will be a `400 Bad Request` in the response and so on.